draft-06/hyper-schema

```diff
@@ -0,0 +1,135 @@
+                    "type": "string"
+                },
+                "encType": {
+                    "description": "The media type in which to submit data along with the request",
+                    "type": "string",
+                    "default": "application/json"
+                },
+                "schema": {
+                    "description": "Schema describing the data to submit along with the request",
+                    "allOf": [ { "$ref": "#" } ]
+                }
+            }
+        }
+    },
+    "allOf": [ { "$ref": "http://json-schema.org/draft-06/schema#" } ],
+    "properties": {
+        "additionalItems": {
+            "anyOf": [
+                { "type": "boolean" },
+                { "$ref": "#" }
+            ]
+        },
+        "additionalProperties": {
+            "anyOf": [
+                { "type": "boolean" },
+                { "$ref": "#" }
+            ]
+        },
+        "dependencies": {
+            "additionalProperties": {
+                "anyOf": [
+                    { "$ref": "#" },
+                    { "type": "array" }
+                ]
+            }
+        },
+        "items": {
+            "anyOf": [
+                { "$ref": "#" },
+                { "$ref": "#/definitions/schemaArray" }
+            ]
+        },
+        "definitions": {
+            "additionalProperties": { "$ref": "#" }
+        },
+        "patternProperties": {
+            "additionalProperties": { "$ref": "#" }
+        },
+        "properties": {
+            "additionalProperties": { "$ref": "#" }
+        },
+        "allOf": { "$ref": "#/definitions/schemaArray" },
+        "anyOf": { "$ref": "#/definitions/schemaArray" },
+        "oneOf": { "$ref": "#/definitions/schemaArray" },
+        "not": { "$ref": "#" },
+
+        "base": {
+            "description": "URI Template resolved as for the 'href' keyword in the Link Description Object.  The resulting URI Reference is resolved against the current URI base and sets the new URI base for URI references within the instance.",
+            "type": "string"
+        },
+        "links": {
+            "type": "array",
+            "items": { "$ref": "#/definitions/linkDescription" }
+        },
+        "media": {
+            "type": "object",
+            "properties": {
+                "type": {
+                    "description": "A media type, as described in RFC 2046",
+                    "type": "string"
+                },
+                "binaryEncoding": {
+                    "description": "A content encoding scheme, as described in RFC 2045",
+                    "type": "string"
+                }
+            }
+        },
+        "readOnly": {
+            "description": "If true, indicates that the value of this property is controlled by the server.",
+            "type": "boolean",
+            "default": "false"
+        }
+    },
+    "links": [
+        {
+            "rel": "self",
+            "href": "{+id}"
```

Comment:
It's a URI template: https://tools.ietf.org/html/rfc6570#section-3.2.3 and this link says that the meta-schema document itself lives at the URI of its own "$id", which (except for having left out the "$") is correct.

But now it should be "{+$id}" except that "$" is excluded from template variable expressions and reserved for pre-processors, so I guess we have to go back to saying that we have a preprocessor that just allows "$"?  Ugh.  Not enough caffeine for this.

I'll have to submit a PR on the other repo when I can figure out how to handle this, and after that's approved update the PR here.